source/yats/include/yats/TaskConfigurator.h

```diff
@@ -82,7 +82,9 @@ class TaskConfigurator : public AbstractTaskConfigurator
 	std::enable_if_t<Index < Helper::ParameterCount> parseInputParameter()
 	{
 		using currentInput = std::tuple_element_t<Index, typename Helper::WrappedInput>;
-		m_inputs.insert(std::make_pair(currentInput::ID, InputConnector(this)));
+		
+		m_inputs.emplace(std::make_pair(currentInput::ID,
```

Comment:
When using emplace you can probably omit the std::make_pair.

Reply:
Shouldn't the slicing problem be solved first?

Reply:
Yes.